### PR TITLE
🐛 Fix bug with spamming the case endpoint

### PR DESCRIPTION
**Pull request by Sven65**
## Explain the changes you’ve made

I've made a small bug fix to stop `CaseOverview` from spamming the `/cases` endpoint.
See #CU-1pkm57y for more details.

## Explain why these changes are made

We don't want to spam the endpoint.

## Explain your solution

As we were relying on a callback hook with a dependency on the `fetchCases` function that comes from context, this would update every time we fetched cases, meaning the hook would run again, triggering the new function and the hook into an infinite loop.

To remedy it, I've simply removed the dependency from the hook.

## How to test

Concrete example:

1. Checkout this branch
2. Enable network inspect for your dev tools
3. See that it doesn't send 40 billion requests a second

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1pkm57y Bugg: Caseoverview-tabben hämtar cases orimligt många gånger](https://app.clickup.com/t/1pkm57y)

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/fedfce3fd8695f238dea9cb00bd37ce7/pull-request/600)

File: source/screens/caseScreens/CaseOverview.tsx [hunk 1]
```diff
@@ -356,7 +356,7 @@ function CaseOverview(props): JSX.Element {
       // since we cannot react to changes as of now.
       const delayBeforeFetch = 4000;
       wait(delayBeforeFetch).then(fetchCases).catch(onFailedToFetchCases);
-    }, [fetchCases])
```
Comment (jlarsson) [inline source/screens/caseScreens/CaseOverview.tsx:359]: This probably fixes the issue , but overall, the combined logic in CaseOverView.tsx is convoluted and should be refactored.

**Review by jlarsson — APPROVED** [review]

File: source/screens/caseScreens/CaseOverview.tsx (hunk 1 above)
Comment (Sven65) [reply source/screens/caseScreens/CaseOverview.tsx:359]: I believe the refactoring is out of the scope of this PR and should be it's own task.

**Review by dannil76 — APPROVED** [review]
looks good

**Pull request merged**